src/utils/src/lib.rs

```diff
@@ -18,3 +23,30 @@ macro_rules! debug {
         ()
     };
 }
+
+// -----------------------------------------------------------------------------
+/// Enum containing error types.
+pub enum Error {
```

Comment:
I find this name to be too generic for having it in the lib.rs of the utils module. I would propose to rename it `FifoError`.